### DISCUSSION TITLE
Code QL error for Java

### DISCUSSION THREAD

**Discussion by pennywise-01** [community/community#29492] [Code Security]
Hi I am new with Code QL and I have a problem to set workflow for java. The error showed like this:
![image](https://user-images.githubusercontent.com/54774679/184607080-8bd7f6b3-ee22-46c2-9a59-f367a42854d7.png)

I read the GitHub documentation but did not understand anything from it, can anyone help me?

**Comment by sj** [comment]
If this is on a public repository: could you post a link to the full Actions log please? That will help us understand the problem better.

What I can tell from the screen shot is that CodeQL either (a) wasn't able to automatically determine how to successfully compile your Java code or (b) the compilation was successful, but no Java source code was seen during it.

Option (a) is most likely, which means that you should probably tell CodeQL how to compile your source code. [Here are some instructions on how to configure that](https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/configuring-the-codeql-workflow-for-compiled-languages#adding-build-steps-for-a-compiled-language).

**Reply by pennywise-01** [reply]
Hi sj, thank you for replying here is the link https://github.com/Hafzan-250601/bookstore/runs/7834785716?check_suite_focus=true. I am not sure if this is the right link, hope it will help you.

**Reply by smowton** [reply]
This doesn't work because your build really doesn't build any .java files. This is because your Maven compiler plugin setup says Java source is found under `src`, but there is no such directory.

I found it works if I move the packages from `OnlineBookStore` into `src/main/java`.

**Reply by pennywise-01** [reply]
It works! Thank youuuu so much
